packages/scripts/config/webpack.config.js

```diff
@@ -36,9 +37,8 @@ if ( ! browserslist.findConfig( '.' ) ) {
 }
 const hasReactFastRefresh = hasArgInCLI( '--hot' ) && ! isProduction;
 
-const copyWebpackPatterns = process.env.WP_COPY_PHP_FILES_TO_DIST
-	? '**/{block.json,*.php}'
-	: '**/block.json';
+// Get paths of the `render` props included in `block.json` files
+const renderPaths = getRenderPropPaths();
```

Comment:
I use wordpress-scripts in our build system, and it doesn't work anymore. The whole thing is more complex, but basically, there is what it does: https://github.com/loxK/wordpress-scripts-test

```javascript
import wpWebpackConfig from '@wordpress/scripts/config/webpack.config.js'
import CopyWebpackPlugin from 'copy-webpack-plugin'
import {resolve} from "path";

export default {
    ...wpWebpackConfig,
    entry: './dir/file.js',
    output: {
        filename: 'file.js',
        path: resolve( process.cwd(), 'assets/js' ),
    },
    plugins: [
        ...(( wpWebpackConfig?.plugins || [] ).filter(plugin => ! (plugin instanceof CopyWebpackPlugin))),

    ]
}

```


```node
» npm run build                               

> build
> webpack

[webpack-cli] Failed to load '/home/www/test/webpack.config.js' config
[webpack-cli] TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at new NodeError (node:internal/errors:387:5)
    at validateString (node:internal/validators:121:11)
    at Object.join (node:path:1172:7)
    at fromProjectRoot (/home/www/test/node_modules/@wordpress/scripts/utils/file.js:13:7)
    at hasProjectFile (/home/www/test/node_modules/@wordpress/scripts/utils/file.js:16:14)
    at getRenderPropPaths (/home/www/test/node_modules/@wordpress/scripts/utils/config.js:312:9)
    at Object.<anonymous> (/home/www/test/node_modules/@wordpress/scripts/config/webpack.config.js:41:21)
    at Module._compile (node:internal/modules/cjs/loader:1126:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1180:10)
    at Module.load (node:internal/modules/cjs/loader:1004:32) {
  code: 'ERR_INVALID_ARG_TYPE'
}
```

Any clue on what to do to make it work again ?

Reply:
Hey @loxK. I've just cloned your repository, done a `npm install` (using node 16) and `npm run build`, and it worked fine.

I'm not sure what may be going on. Could you please try removing the `node_module` folder and running `npm install` again? Or giving us more precise instructions to reproduce the problem?

Reply:
You need to upgrade to @wordpress/scripts to 24.1.0 by editing packages.json and `npm i`, in the repository the wordpress-scripts version is fixed to 24.0.0.

Reply:
Gotcha. 

For some reason, `process.env.WP_SRC_DIRECTORY` is `undefined` in your configuration. I'll investigate why. 

It wasn't failing before because you overwrote the entry points (which also access `process.env.WP_SRC_DIRECTORY`).

Reply:
Ok, it's because you're using `"build": "webpack"` and not `"build": "wp-scripts build"` in your `package.json`.

`wp-scripts build` initializes `process.env.WP_SRC_DIRECTORY`: https://github.com/WordPress/gutenberg/blob/0a0598f147f3b413e6b223eb5acdb07bed153b29/packages/scripts/scripts/build.js#L27

Reply:
I've opened a PR to clarify that part in the docs: 

- https://github.com/WordPress/gutenberg/pull/44348

Reply:
I don't want to use wp-scripts, I want to extend the webpack config and customize input and output directories and file name. It is part of a more complex build system.

Reply:
To be more precise, I use it with [webpack stream](https://www.npmjs.com/package/webpack-stream)

Reply:
Then, you need to take a look at what `wp-scripts` is doing before loading webpack and replicate it in your setup. For now, you may get away initilizing `process.env.WP_SRC_DIRECTORY` to the correct value. But expect it to keep breaking in the future as the webpack configuration expects to be used within the `wp-scripts` commands.

Reply:
How can I use multiple webpack configs with wp scripts ? You made it overcomplicated to extend, that is a regression from a developer side view  ...

Reply:
> How can I use multiple webpack configs with wp scripts?

Webpack [accepts multiple configurations](https://webpack.js.org/configuration/configuration-types/#exporting-multiple-configurations) if you export them in an array.

This is an example using `wp-scripts`: https://github.com/WordPress/block-hydration-experiments/blob/main-full-vdom-hydration/webpack.config.js

Reply:
What is the purpose of process.env.WP_SRC_DIRECTORY ? If I want to extend the webpack config to feed a single file, it look like a nonsense.

Reply:
I tried setting `process.env.WP_SRC_DIRECTORY = './dir1'`, it still fails to build 😬

Reply:
I'm sorry, but I don't know the answer. You need to find a way to switch back to `wp-scripts`, or you'd have to take a look at its internals to understand what it's doing before it loads webpack to do that yourself. You're also welcome to open a PR if you think there's something broken inside `wp-scripts` 🙂

Reply:
Damn breaking change in a minor release version number. Hours of work ahead ...
That sucks.

Reply:
process.env.WP_SRC_DIRECTORY has be back to being optional, like it has always been, and it would be perfect.

Reply:
Extending webpack is not covered with semver, especially when not used in conjunction with `wp-scripts`. 

[From the docs](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#provide-your-own-webpack-config):

> Future versions of this package may change what webpack and Babel plugins we bundle, default configs, etc.

Reply:
By the way, if you have needs that `wp-scripts` don't support yet and you think they'd be useful for a wide range of people, I encourage you to open an issue/PR to add those improvements to `wp-scripts` 🙂

I also encourage you to open a PR if you think there's something broken or that could be done in a better way inside `wp-scripts`.

Reply:
For reference, @loxK opened a PR to change the way the webpack config access the value of `process.env.WP_SRC_DIRECTORY`:

- https://github.com/WordPress/gutenberg/pull/44367